src/util/chunk_fs.js

```diff
@@ -28,54 +28,50 @@ class ChunkFS extends stream.Transform {
         this._total_num_buffers = 0;
     }
 
-    _transform(chunk, encoding, callback) {
+    async _transform(chunk, encoding, callback) {
         try {
-            this._process_chunk(chunk, callback);
+            if (this.MD5Async) await this.MD5Async.update(chunk);
+            stats_collector.instance(this.rpc_client).update_namespace_write_stats({
+                namespace_resource_id: this.namespace_resource_id,
+                size: chunk.length,
+                count: this.count
+            });
+            this.count = 0;
+            while (chunk && chunk.length) {
+                const available_size = config.NSFS_BUF_SIZE - this.q_size;
+                const buf = (available_size < chunk.length) ? chunk.slice(0, available_size) : chunk;
+                this.q_buffers.push(buf);
+                this.q_size += buf.length;
+                if (this.q_size === config.NSFS_BUF_SIZE) await this._flush_buffers();
```

Comment:
I'd use >= just being defensive
```suggestion
                if (this.q_size >= config.NSFS_BUF_SIZE) await this._flush_buffers();
```